docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Native files in .NET packages
+description: How to pack native libraries in .NET packages
+author: zivkan
+ms.author: zivkan
+ms.date: 07/15/2023
+ms.topic: conceptual
+---
+
+# Including native libraries in .NET packages
+
+This document explains how to create packages that work for projects targeting .NET (5+), where a .NET managed assembly uses [Platform Invoke (P/Invoke)](/dotnet/standard/native-interop/pinvoke) from a .NET managed language project, with the .NET runtime using [probing paths](/dotnet/core/dependency-loading/default-probing) to load and locate the native library.
+
+To support other scenarios, your best option is to package your own [MSBuild props and targets files](../concepts/MSBuild-props-and-targets.md), and handle build and publish targets yourself.
+By not using NuGet and the .NET runtime's built in feature,  you need to be sufficiently knowledgeable in MSBuild and the build systems for the relevant project types.
+.NET Framework ASP.NET projects in particular work quite differently with regards to many other project types, but .NET Framework (non-SDK style) projects may also not work the same as SDK style projects.
```

Comment:
todo if I can muster up the motivation: check if SDK style projects targeting .NET Framework also work

more todo if time & motivation can be found: What really works for legacy csproj PackageReference? I'm sure that NuGet will do asset selection if the RIDs match, but from memory there is do `deps.json` support before .NET Core.

Reply:
after a quick test, an SDK style project targeting a .NET Framework TFM will by default target the "AnyCPU" platform, and will copy all the runtime files into a `runtimes/` directory, but will fail at runtime since it doesn't have the `.deps.json` file (I'm sure because netfx doesn't support it, not because the SDK has a bug). [I can't find docs on how P/Invoke locates native libraries on .NET Framework](https://github.com/dotnet/docs/issues/36539). The `AppDomain` class has `AssemblyLoad` and `AssemblyResolve` events, but since Assembly usually refers specifically to a managed library, I assume it doesn't work for unmanaged libraries.

However, using VS to set the platform to either x86 or x64, it works.

So, this is more of a "using packages" issue, not a "creating packages" issue, but I'll have a think later about what to add to this docs page.